### ISSUE TITLE
Crash on startup in host OS when processes inside Docker containers have uids that don't exist in the host OS

### ISSUE THREAD

**Issue by marshallpierce** [nicolargo/glances#845]
#### Description

When processes are running inside a Docker container, they can have uids in the host OS that do not exist according to /etc/passwd.

I'm running glances 2.5.1 on python 2.7.1, and I get the following when launching glances:

```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python2.7/glances", line 9, in <module>
    load_entry_point('Glances==2.5.1', 'console_scripts', 'glances')()
  File "/usr/lib64/python2.7/site-packages/glances/__init__.py", line 122, in main
    args=core.get_args())
  File "/usr/lib64/python2.7/site-packages/glances/core/glances_standalone.py", line 68, in __init__
    self.stats.update()
  File "/usr/lib64/python2.7/site-packages/glances/core/glances_stats.py", line 156, in update
    self._plugins[p].update()
  File "/usr/lib64/python2.7/site-packages/glances/plugins/glances_processcount.py", line 58, in update
    glances_processes.update()
  File "/usr/lib64/python2.7/site-packages/glances/core/glances_processes.py", line 603, in update
    standard_stats=self.max_processes is None)
  File "/usr/lib64/python2.7/site-packages/glances/core/glances_processes.py", line 569, in __get_process_stats
    procstat = self.__get_mandatory_stats(proc, procstat)
  File "/usr/lib64/python2.7/site-packages/glances/core/glances_processes.py", line 383, in __get_mandatory_stats
    'name', 'cpu_times'], ad_value=''))
  File "/usr/lib64/python2.7/site-packages/psutil/__init__.py", line 414, in as_dict
    ret = attr()
  File "/usr/lib64/python2.7/site-packages/psutil/__init__.py", line 566, in username
    return pwd.getpwuid(self.uids().real).pw_name
KeyError: 'getpwuid(): uid not found: 999'
```

999 is not a user that exists in the host OS, but I do have a process with uid 999 that is running inside a docker container:

```
% ps aux | grep 999
999      28744  0.0  0.0  37208  3728 ?        Ssl  Apr05   1:18 redis-server *:6379
```

#### Versions

* Glances (glances -V): 2.5.1
* PSutil (glances -V): 2.1.3
* Operating System (lsb_release -a): Gentoo x64, current.

#### Logs

I don't see anything useful in the logs, just the versions above.

**Comment by nicolargo** [comment]
Hi @marshallpierce thanks for the report.

Can you send us a procedure to reproduce this issue ?

**Comment by marshallpierce** [comment]
Sure. Run a docker container (this one is quick to download):
```
docker run --name local-redis -d redis:alpine
```

That will probably map redis to a uid that isn't in your system. If it does, try other docker containers.

Then run glances.

**Comment by nicolargo** [comment]
Looks like a PsUtil issue. Two actions should be taken:

1. Open an issue on the PsUtil Github
2. Try / except the line 383 (glances_processes.py)

**Comment by dflock** [comment]
https://github.com/giampaolo/psutil/issues/547

**Comment by nicolargo** [comment]
Corrected in PsUtil 2.2.0 or higher
